jim.h

```diff
@@ -333,6 +336,12 @@ void jim_array_begin(Jim *jim)
 void jim_array_end(Jim *jim)
 {
     if (jim->error == JIM_OK) {
+        Jim_Scope *scope = jim_current_scope(jim);
```

Comment:
`jim_current_scope` may return `NULL` if the stack is empty. This must be checked.